compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java

```diff
@@ -95,16 +95,13 @@ public static Path getLibDir() {
      * @return settings object
      */
     public static Settings readSettings() {
-        if (settings == null) {
-            String tomlFilePath = RepoUtils.createAndGetHomeReposPath().resolve(ProjectDirConstants.SETTINGS_FILE_NAME)
-                                           .toString();
-            try {
-                settings = SettingsProcessor.parseTomlContentFromFile(tomlFilePath);
-            } catch (IOException e) {
-                settings = new Settings();
```

Comment:
So there seems to be an unused settings variable at this class level?

Reply:
Removed the unused variable with 9a9ccfc